datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -145,4 +145,26 @@ impl PhysicalExpr for ScalarFunctionExpr {
         let fun = self.fun.as_ref();
         (fun)(&inputs)
     }
+
+    fn children(&self) -> Vec<Arc<dyn PhysicalExpr>> {
+        self.args.clone()
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        children: Vec<Arc<dyn PhysicalExpr>>,
+    ) -> Result<Arc<dyn PhysicalExpr>> {
+        Ok(Arc::new(ScalarFunctionExpr::new(
+            &self.name,
+            self.fun.clone(),
+            children,
+            self.return_type(),
+        )))
+    }
+}
+
+impl PartialEq<dyn Any> for ScalarFunctionExpr {
+    fn eq(&self, _other: &dyn Any) -> bool {
+        false
```

Comment:
This seems incorrect / unfinshed -- shouldn't it be checking the contents of the `other`?

Reply:
I leave it here because I don't know how to compare `fun` member, its type is the boxed Closure. 
Or can we skip the `fun` member and just compare `name`, `args` and `return_type` ?

```

pub struct ScalarFunctionExpr {
    fun: ScalarFunctionImplementation,
    name: String,
    args: Vec<Arc<dyn PhysicalExpr>>,
    return_type: DataType,
}


pub type ScalarFunctionImplementation =
    Arc<dyn Fn(&[ColumnarValue]) -> Result<ColumnarValue> + Send + Sync>;
```

Reply:
Comparing name, args and return type seems like a pretty good comparison to me

Reply:
Ok